torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -398,19 +411,63 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
       : bdimx + (device_warp_size - bdimx % device_warp_size);
 
   bool pad_bdimx = bdimx > 16 &&
-      padded_bdimx * bdimy * bdimz <
-          (int64_t)at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;
+      padded_bdimx * bdimy * bdimz < (int64_t)dev_prop->maxThreadsPerBlock;
 
   pad_bdimx = pad_bdimx &&
       bdimx * inner_reduction_unroll_factor != inner_most_dimension_numel;
 
+  int64_t nvrtc_register_per_thread = LaunchParams::UNINITIALIZED_VAL;
+  const int blocksPerKernel = godim;
+  if (vectorize && blocksPerKernel > device_multiprocessor_count &&
```

Comment:
Comment why these conditions

Reply:
Re: vectorization, please mention we've seen unexpectedly high register counts with vectorization factor less than 4, which would make the below estimate inaccurate.

A TODO item would be to support the non vectorized case. We could do more conservative estimate, but at that point it might be better to use shmem. Not for this PR, but something we may need to work on later.